src/dict.c

```diff
@@ -333,8 +333,8 @@ int dictRehash(dict *d, int n) {
     unsigned long s1 = DICTHT_SIZE(d->ht_size_exp[1]);
     if (dict_can_resize == DICT_RESIZE_FORBID || !dictIsRehashing(d)) return 0;
     if (dict_can_resize == DICT_RESIZE_AVOID && 
-        ((s1 > s0 && s1 / s0 < dict_force_resize_ratio) ||
-         (s1 < s0 && s0 / s1 < dict_force_resize_ratio)))
+        ((s1 > s0 && (double)s1 / s0 < dict_force_resize_ratio) ||
+         (s1 < s0 && (double)s0 / s1 < dict_force_resize_ratio)))
```

Comment:
We can avoid the integer division if we multiply both sides with the dividend.

```suggestion
        ((s1 > s0 && s1 < dict_force_resize_ratio * s0) ||
         (s1 < s0 && s0 < dict_force_resize_ratio * s1)))
```

Reply:
Hi, both s0 and s1 here is the size of the dict and must be the powers of 2. The division result can't be double or float number. Do I miss something?

Reply:
yean, let see which one is better, i am ok with both

Reply:
integer multiplication is probably faster than double division on every platform, if performance is relevant here